### ISSUE TITLE
Standalone Installation on Windows is not working because FFmpeg installation failed

### ISSUE THREAD

**Issue by Maxwell77HH** [erew123/alltalk_tts#350]
Hi, 
I tried to install alltalk_tts like described. After I started the atsetup.bat everything works great till the point at which FFmpeg should be installed. I get the error:

FFmpeg installation failed, errorlevel was 1. This could be caused by:

     1. Internet connection issues/unable to download FFmpeg from Conda's website.
     2. Disk space related issues - Check your disk space on this drive.
     3. Incorrect or missing Conda environment - Restart installation.
     4. Permissions issues - Check you have enough rights on this system.
     5. Firewall or proxy settings blocking access to Conda's servers.
     6. Antivirus or security software interference.
     7. Other issues not mentioned above.
 Do you want to retry the FFmpeg installation? [Y,N]?

Unfortunatley a retry don´t solve the problem. When I hit "N" it tries to install FFmpeg again. To hit "N" is the same like to hit  "Y". There is no way to exit this setup at this point.

**Comment by erew123** [comment]
Hi @Maxwell77HH Maxwell

I've swapped the method it uses to check the Y/N response, which in theory should work, so you can `git pull` down those changes. However, the looping Y/N message is not the actual issue here. The actual issue is that for some reason, your system cannot reach Conda Forge, which is where Conda store their downloadable files.

The line in the code is running this https://github.com/erew123/alltalk_tts/blob/alltalkbeta/atsetup.bat#L451

FYI, you can always exit a batch file by pressing `CTRL+C` together and that will quit out any batch file.

As for the actual issue, I would say that 90% of the time this type of problem is usually an intermittent internet connection issue. So you may well just wish to re-try and see what happens.

I would `Delete AllTalk's custom Python environment` and then `Install AllTalk as a Standalone Application`

![image](https://github.com/user-attachments/assets/40e312ec-c83e-4a12-b3fd-093674364aa5)

---

### Generally testing your system can reach Conda's website.

You can do a general test at a command prompt, to check if your system can reach Conda's site by running the following (assuming CURL is installed on your system) `curl -I https://conda.anaconda.org/conda-forge`. A healthy response would look like this:

![image](https://github.com/user-attachments/assets/54362c90-09ec-4f83-af32-d4aed5d01a7d)

This **only** proves that at that moment in time, your system can reach out across the internet to Conda's website. It doesn't tell you that conda's website is working correctly. Obviously any other type or error message would indicate a connectivity issue somewhere OR Conda's website is down.

---

### Looking for Error codes in the installation routine

When you re-run the installation, if it errors again, you **should** have an error code something like one of the following, which we would need to figure out what exactly is the issue on your installation:

**Conda Forge unreachable:** If Conda cannot connect to Conda Forge, you might see an error like:

`CondaHTTPError: HTTP 000 CONNECTION FAILED for url <url-to-conda-forge>`

This indicates network issues or that Conda Forge is not reachable from your system.

**Dependency Conflicts:** If there are dependency conflicts, Conda will show a message like:

`PackagesNotFoundError: The following packages are not available from current channels:`
`  - ffmpeg`

**Insufficient Disk Space:** If the system runs out of disk space, you might see:

`OSError: [Errno 28] No space left on device`

**Conda Environment Issues:** If there's an issue with the Conda environment (e.g., it's broken or missing necessary packages), you might see errors like:

`CondaEnvironmentError: Could not find environment <environment_name>.`

**Permission Issues:** If Conda lacks necessary permissions, you might see:

`PermissionError: [Errno 13] Permission denied: '<path>'`

---

So in summary:

- Open a command prompt and navigate to the `alltalk_tts` folder then `git pull` to get the updated atsetup.bat
- Delete the Python Environment
- Run the setup again to see if the problem is resolved, indicating it was a transient issue, OR look for any error message that indicates why it failed that step.

Thanks

**Comment by Maxwell77HH** [comment]
Hi, thank you very much for your response. I tested it with the command "curl -I https://conda.anaconda.org/conda-forge" if I can reach the anaconda website. Everything is fine. 

HTTP/1.1 302 Found
Date: Wed, 25 Sep 2024 05:22:25 GMT
Content-Type: text/html; charset=utf-8
Connection: keep-alive
Location: https://anaconda.org/conda-forge/repo?type=conda&label=main
CF-Ray: 8c88938d6f3062ff-HAM
CF-Cache-Status: DYNAMIC
Strict-Transport-Security: max-age=0
content-security-policy: frame-ancestors 'self';
referrer-policy: no-referrer
x-content-type-options: nosniff
x-download-options: noopen
Set-Cookie: __cf_bm=CRvwv4PXoXZasjhfsVmf3pgRBY2TKvS2NDK7NgfItl4-1727241745-1.0.1.1-gOICpfm.auPj8ERvizTO5kPpZ5BrbtpWbvG2bchlOZetFNS7LEN96QmXGrmKYuSNZGTpQPyydo4XoEbP6BZ40mP1BOEyIuoZ.uSXfIWm.u8; path=/; expires=Wed, 25-Sep-24 05:52:25 GMT; domain=.anaconda.org; HttpOnly; Secure; SameSite=None
Server: cloudflare

But FFmpeg can´t be installed when I run the installer. But for some reason I found the FFmpeg.exe in the install directory.  After I canceled the FFmpeg install routine in the installer it quits. Then I opened the admin.html file. I get the interface. But all engines and models are not available. I can´t choose anything from the dropdown menus. What could cause this problem?

**Comment by erew123** [comment]
Hi @Maxwell77HH 

I cannot say without an error code, other than getting an error code such as what I suggested above in the looking for error codes.

As an alternative, what I can offer is a full copy/paste, step by step guide to performing a manual installation.

Please see [MANUAL Windows Setup - Standalone Installation here](https://github.com/erew123/alltalk_tts/tree/alltalkbeta?tab=readme-ov-file#pythonpip-has-recently-made-changes-to-the-package-installer-and-i-haveam-doing-what-i-can-to-keep-the-requirements-files-working-however-this-is-a-complicated-time-to-say-the-least-as-some-external-requirements-change-and-this-can-cause-installation-issues-please-raise-a-ticket-if-you-have-an-error-that-a-module-is-missing)

You can go through those steps and look for error messages occurring at each step.

I've tested through both that manual process and also atsetup.bat today and have been unable to create an error for myself, so whatever the issue is, its unique to your computer or internet connection. 

Going through those steps manually should provide an error code if one exists.

Thanks

**Comment by brehiner** [comment]
I solved with this code in terminal: conda config --set channel_priority flexible

**Comment by tre4b** [comment]
> I solved with this code in terminal: conda config --set channel_priority flexible

Just came here to say the same thing with my installation after working through the batch file trying to figure out why it would not install.  I tried it manually and though the install manually still failed I had used the above command as part of that and when I tried again it all worked....
